File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();

        for (int i = 0; i < strs.length; i++) {
            char[] temp = strs[i].toCharArray();
            Arrays.sort(temp);
            String curr = new String(temp);

            if (!map.containsKey(curr)) {
                map.put(curr, new ArrayList<>());
            }

            map.get(curr).add(strs[i]);
        }
        return new ArrayList<>(map.values());
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        seen = {}
        for word in strs:
            # sorted returns list of characters
            sorted_word = "".join(sorted(word))
            if sorted_word in seen:
                seen[sorted_word].append(word)
            else:
                seen[sorted_word] = [word]

        res = []
        for word in seen:
            temp = seen[word]
            res.append(temp)
        
        return res
